static/app/components/modals/inviteMissingMembersModal/index.tsx

```diff
@@ -0,0 +1,342 @@
+    teamSlugs: new Set<string>(),
+    externalId: member.externalId,
+    selected: false,
+  }));
+  const [memberInvites, setMemberInvites] =
+    useState<MissingMemberInvite[]>(initialMemberInvites);
+  const [inviteStatus, setInviteStatus] = useState<InviteStatus>({});
+  const [sendingInvites, setSendingInvites] = useState(false);
+  const [complete, setComplete] = useState(false);
+
+  const api = useApi();
+
+  if (!memberInvites || !organization.access.includes('org:write')) {
+    return null;
+  }
+
+  const setRole = (role: string, index: number) => {
+    setMemberInvites(
+      memberInvites.map((member, i) => {
+        if (i === index) {
+          member.role = role;
+        }
+        return member;
+      })
+    );
+  };
+
+  const setTeams = (teamSlugs: string[], index: number) => {
+    setMemberInvites(
+      memberInvites.map((member, i) => {
+        if (i === index) {
+          member.teamSlugs = new Set(teamSlugs);
+        }
+        return member;
+      })
+    );
+  };
+
+  const selectAll = (checked: boolean) => {
+    const selectedMembers = memberInvites.map(m => {
+      m.selected = checked;
+      return m;
+    });
+    setMemberInvites(selectedMembers);
+  };
+
+  const checkboxToggle = (checked: boolean, index: number) => {
+    const selectedMembers = [...memberInvites];
+    selectedMembers[index].selected = checked;
+    setMemberInvites(selectedMembers);
+  };
+
+  const renderStatusMessage = () => {
+    if (sendingInvites) {
+      return (
+        <StatusMessage>
+          <LoadingIndicator mini relative hideMessage size={16} />
+          {t('Sending organization invitations\u2026')}
+        </StatusMessage>
+      );
+    }
+
+    if (complete) {
+      const statuses = Object.values(inviteStatus);
+      const sentCount = statuses.filter(i => i.sent).length;
+      const errorCount = statuses.filter(i => i.error).length;
+
+      const invites = <strong>{tn('%s invite', '%s invites', sentCount)}</strong>;
+      const tctComponents = {
+        invites,
+        failed: errorCount,
+      };
+
+      return (
+        <StatusMessage status="success">
+          <IconCheckmark size="sm" />
+          {errorCount > 0
+            ? tct('Sent [invites], [failed] failed to send.', tctComponents)
+            : tct('Sent [invites]', tctComponents)}
+        </StatusMessage>
+      );
+    }
+
+    return null;
+  };
+
+  const sendMemberInvite = async (invite: MissingMemberInvite) => {
+    const data = {
+      email: invite.email,
+      teams: [...invite.teamSlugs],
+      role: invite.role,
+    };
+
+    try {
+      await api.requestPromise(`/organizations/${organization?.slug}/members/`, {
+        method: 'POST',
+        data,
+      });
+    } catch (err) {
+      const errorResponse = err.responseJSON;
+
+      // Use the email error message if available. This inconsistently is
+      // returned as either a list of errors for the field, or a single error.
```

Comment:
😅 that's not good API response. I add it to my long list.

Reply:
this is the same comment in the current invite members modal too